lib/auth/auth_with_roles.go

```diff
@@ -1536,11 +1536,20 @@ func (a *ServerWithRoles) ListUnifiedResources(ctx context.Context, req *proto.L
 	}
 
 	unifiedResources, nextKey, err := a.authServer.UnifiedResourceCache.IterateUnifiedResources(ctx, func(resource types.ResourceWithLabels) (bool, error) {
-		if err := resourceChecker.CanAccess(resource); err != nil {
-			if trace.IsAccessDenied(err) {
-				return false, nil
+		switch r := resource.(type) {
+		case types.SAMLIdPServiceProvider:
```

Comment:
i'm not sure i'm reading this right, but if err does != nil, won't it still try to `MatchResourceByFilters`? (especially the second case returns false )